platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -3960,6 +4001,8 @@ int odp_tm_node_context_set(odp_tm_node_t tm_node, void *user_context)
 void odp_tm_queue_params_init(odp_tm_queue_params_t *params)
 {
 	memset(params, 0, sizeof(odp_tm_queue_params_t));
+
+	params->ordered_enqueue = true;
```

Comment:
This is only part of the implementation. Now that support for unordered enqueue was added, the implementation should look at the ordered_enqueue parameter and skip the ordering step (i.e. the _odp_sched_fn->order_lock() and the unlock call) in tm_enqueue().

Reply:
Ok. I can try making those two locks conditional on this flag of unordered enqueue. Is that sufficient ?
If not, then can you or matias do the change as I don't have much internal understanding of linux-generic.

Another approach could be that, since maintaining ordering even when ordered_enqueue is false is not functionally incorrect but just has performance implications, we can leave the implementation as it is and you guys can optimize it later ?